### DISCUSSION TITLE
More data types

### DISCUSSION THREAD

**Discussion by inputsh** [Budibase/budibase#1041] [Ideas]
Thank you for the awesome app! Looking forward to future updates, and would love to see more simple data types added in. Examples include:

* **Time duration (HH:mm:ss)** would be for any sort of time tracking (examples: sleep time, exercise time, average website session etc). 
* **Currency**, which would be similar to numbers, but always with decimals and with a couple of pre-built signs (USD, EUR, BTC, etc) and a user-defined ones for lesser used currencies (XPR, HRK, etc). Number field could technically be extended to include decimals and a suffix as a workaround.
* **ID**, uniquely generated per row. Would be useful in myriads of ways, but mostly to have some differentiation between two records that happen to have identical values.
* **URL** would be useful to keep some lists, for example, articles, books, videos, etc.
* **Multi-select**. The option currently works as a category (one row can only belong to one option). This would be opposite, with options working more as tags (one row can have multiple tags).
* ...

**Comment by DaFrenchFrog** [comment]
I'm beginning on budibase and support both the app and this post.

**Reply by gvlx** [reply]
Hi,

+1 for Currency but you need to set locale (by field) for this work correctly (thousands and decimal separators, currency symbol and placement, etc.)
+ 1 for Time/date/timestamp, but the same locale problems exist

Multi-select you already have, but maybe you are talking about new ways to present the field.
I don't know if we can presently add things dinamically (that is, write-in new values).

**Reply by DaFrenchFrog** [reply]
Thx, I was thinking (and actually in need) of a duration type which would avoid calculations through date.
